uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Page/UserMenu.php

```diff
@@ -1,102 +1,13 @@
 <?php
 
-use CRM_HRCore_CMSData_PathsFactory as CMSPathsFactory;
-
 class CRM_HRCore_Page_UserMenu extends CRM_Core_Page {
 
   /**
-   * The contact data used to build the menu
-   *
-   * @var array
-   */
-  private $contactData = [];
-
-  /**
-   * An instance of a class implementing the CRM_HRCore_CMSData_PathsInterface
-   *
-   * @var CRM_HRCore_CMSData_PathsInterface
-   */
-  private $cmsPaths;
-
-  /**
    * {@inheritdoc}
    */
   public function run() {
-    $this->instantiateCmsPaths();
-
-    $this->assign('username', $this->contactData()['display_name']);
-    $this->assign('image', $this->getUserImagePath());
-    $this->assign('editLink', $this->cmsPaths->getEditAccountPath());
-    $this->assign('logoutLink', $this->cmsPaths->getLogoutPath());
+    $this->assign('userMenuMarkup', (new CRM_HRCore_UserMenuMarkup())->getMarkup());
```

Comment:
In PHP, you can have a string representation of a class. That is, if you "echo" an object, PHP will print this string representation. This is done by implementing the `__toString()` method in a class. It means that, if you rename this `getMarkup()` method to `__toString()`, you can change this line to simply:

```php
$this->assign('userMenuMarkup', new CRM_HRCore_UserMenuMarkup());
```

Reply:
will it be understandable what's going on though? Is it common to do it?

Reply:
I think it's common in cases we objects represent UI elements. As for understandability, I think it's fine as this is not a hidden secret of PHP or something, but it might be better to ask other devs about this. Or, if you think it's too much, then it's ok to leave it as it is. I really have no strong opinions on this

Reply:
if it's ok with you i'd leave it as is, especially cause we could have other non-backend devs working on it